### ISSUE TITLE
[Bug][a11y]: Picker with VoiceOver on iOS fails to restore focus to button when listbox closes 

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#3242]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Picker

### Expected behavior

VoiceOver cursor should restore focus to the button in the `<sp-picker>` `shadowDom` after the listbox closes on item selection or triggering the Dismiss button. 

### Actual behavior

After the listbox closes, while the `<sp-picker>` renders as if it has focus, the VoiceOver cursor focus seems to be lost. In this state, trying to navigate to the previous or next element or double tapping to expand the picker again sounds an "earcon" to indicate that nothing happened. In order to find focus the `sp-picker` again, the user is forced to explore by touch.

The issue can also be reproduced with Chrome and Talkback on Android.

### Screenshots

Screen capture demonstrating the issue with VoiceOver on iOS: https://shared-assets.adobe.com/link/994233f8-47ad-4759-46c8-93a74cdd0a9d

### What browsers are you seeing the problem in?

- Safari on iOS with VoiceOver
- Chrome on Android with Talkback

### How can we reproduce this issue?

Using VoiceOver on iOS:

1. Go to https://opensource.adobe.com/spectrum-web-components/components/picker/#sizes
2. Navigate the VoiceOver cursor to the first Picker labelled "Selection type:"
3. Double tap to activate the Picker, expanding the listbox popover or tray.
4. Swipe to focus an option, then double tap to select it.
5. The popover or tray will close, the Picker will render its focus style, but the VoiceOver cursor does not restore focus to the Picker button.
6. Try swiping left, right, up or down to navigate using VoiceOver to the next or previous element. VoiceOver will be unable to navigate and will sound an "earcon".

**Comment by majornista** [comment]
Note that this issue can also be reproduced with Chrome on Android using Talkback.

**Comment by Westbrook** [comment]
We're seeing this issue as no longer present in iOS and Android. There is a related issue around getting back and forth between the options in the Tray, but I'll open a separate issue with specific info around that.
